File: CPP/PrintPi.cpp
//
//  PrintPi.cpp
//  
//
//  Created by Evan Dorn on 4/26/16.
//
//

/*
 * Problem statement
 * 1.) Find PI to the Nth Digit - Enter a number and have the program generate 
 *  PI up to that many decimal places. Keep a limit to how far the program will go.
 */

#include <iostream>
#include <iomanip>
#include <cmath>

#define _USE_MATH_DEFINES           // For older systems in order to access M_PI

int main() {
    int precisionWidth;
    std::cout << "Enter the number of decimal places (up to 50) you want to print Pi: " << " ";
    std::cin >> precisionWidth;
    if(precisionWidth >= 50) {
        std::cout << "You entered " << precisionWidth << ", input is too large. Printing PI at 50 places." << std::endl;
        std::cout << std::setprecision(precisionWidth) << M_PI << std::endl;
    } else {
    std::cout << "PI at " << precisionWidth << " places" << std::endl;
    std::cout << std::setprecision(precisionWidth) << M_PI << std::endl;
    }
    return 0;
}

File: swift/PrintPi.swift
//
//  PrintPi.swift
//  
//
//  Created by Evan Dorn on 5/1/16.
//
//

import Foundation
import Darwin                       // Gives access to C functions and M_PI constant


// Note: M_PI is only valid to 15 digits
func piToNth(n:Int) -> Void {
    if (n < 0 || n > 15) {
        print("Error: Not a valid input range. Enter a number between 1 and 15")
        exit(1)
    }
    let digits = floor((M_PI)*(pow(10.0,Double(n))))
    print((digits/pow(10.0,Double(n))))
}

// FWIW print("Foo", terminator:"") is the prefered way of printing without a newline
print("Enter the number of decimal places (up to 15) you want to print Pi: " + " ")
let input:String? = readLine(stripNewline: true)
let precisionWidth:Int? = Int(input!)
piToNth(precisionWidth!)
